src/org/labkey/targetedms/model/SampleFileQCMetadata.java

```diff
@@ -0,0 +1,50 @@
+package org.labkey.targetedms.model;
+
+import org.apache.commons.lang3.StringUtils;
+import org.jetbrains.annotations.Nullable;
+import org.labkey.targetedms.parser.SampleFile;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class SampleFileQCMetadata extends SampleFile
+{
+    boolean inGuideSetTrainingRange;
+    private Set<Integer> _ignoredMetricIds = Collections.emptySet();
+
+    @Nullable
+    public boolean isIgnoreInQC(int metricId)
+    {
+        return _ignoredMetricIds.contains(metricId) || _ignoredMetricIds.contains(-1);
```

Comment:
Is this correct? I might be understanding incorrectly but presence of -1 indicates that metricId is not present in QCMetricExclusion.

Reply:
I believe it's correct. I just pushed a commit with comments in both the Java code and SQL - can you see if it makes more sense now?

Reply:
yeah it makes sense now with the context. Thanks.